### PR TITLE
[SERVICES-2449] REST endpoints implementing trading view UDF protocol

### PR DESCRIPTION
**Pull request by mad2sm0key**
## Reasoning
- expose REST endpoints returning price candlestick data in a format compatible with the [UDF protocol](https://www.tradingview.com/charting-library-docs/latest/connecting_data/UDF/) of the Trading View charting library 
  
## Proposed Changes
- add TradingViewController exposing 3 endpoints under the `/trading-view` path :
  -`GET /config` : datafeed configuration (supported resolutions, availability of libracry specific features) 
  -`GET /symbols?symbol=<symbol>` : symbol information; this endpoint works with a single token identifier, or with the tokens of a pair separated by a colon (eg `WEGLD-a28c59:UTK-14d57d` or `UTK-14d57d:WEGLD-a28c59`)
  -`GET /history?symbol=<ticker_name>&from=<unix_timestamp>&to=<unix_timestamp>&resolution=<resolution>&countback=<countback>` : the actual bars with price data in the format expected by the library
- add TradingView service for resolving symbols and converting the timeseries data to the expected format

## How to test
- GET request to `/trading-view/config` should return a JSON in this format :
```
{
    supported_resolutions: ['1', '5', '15', '30', '60', '1D', '1W', '1M'],
    supports_group_request: true,
    supports_marks: false,
    supports_search: false,
    supports_timescale_marks: false,
}
```
- GET request to `/trading-view/symbols?symbol=UTK-14d57d` should return a JSON in this format : 
```
{
    ticker: 'UTK-14d57d',
    name: 'UTK',
    type: 'crypto',
    session: '24x7',
    timezone: 'Etc/UTC',
    exchange: 'xExchange',
    minmov: 1,
    pricescale: 100000,
    has_daily: true,
    has_intraday: true,
    has_weekly_and_monthly: true,
    has_empty_bars: true,
    visible_plots_set: 'ohlc',
    supported_resolutions: ['1', '5', '15', '30', '60', '240', '1D', '7D', '1M'],
    data_status: 'streaming',
}
```
- GET request to `/trading-view/history?symbol=UTK-14d57d&resolution=1D&from=1718350507&to=1718637437` should return a JSON in this format :
```
{
  s: "ok",
  t: [
    1718409600,
    1718496000,
    1718582400
  ],
  o: [
    0.12059060143529163,
    0.12129578032794155,
    0.11914486723807885
  ],
  h: [
    0.12129253885032475,
    0.12129578032794155,
    0.11914486723807885
  ],
  l: [
    0.12059060143529163,
    0.11916423955026223,
    0.11914486723807885
  ],
  c: [
    0.12129253885032475,
    0.11917772753150199,
    0.11914486723807885
  ]
}
```

### CODE REVIEW

File: src/modules/trading-view/dtos/bars.response.ts [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    MINUTE_30 = '30',
+    HOUR_1 = '60',
+    HOUR_4 = '240',
+    DAY_1 = '1D',
+    DAY_7 = '7D',
+    MONTH_1 = '1M',
+}
+
+export class BarsResponse {
+    s: string;
```
Comment (bogdan-rosianu) [inline src/modules/trading-view/dtos/bars.response.ts:30]: maybe add comments for each field? the short names aren't descriptive

File: src/modules/trading-view/services/trading.view.service.ts [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+export class TradingViewService {
+    constructor(
+        private readonly analyticsQueryService: AnalyticsQueryService,
+        private readonly tokenService: TokenService,
+        private readonly routerAbi: RouterAbiService,
+    ) {}
+
+    async resolveSymbol(
+        symbol: string,
+    ): Promise<{ ticker: string; name: string }> {
```
Comment (claudiulataretu) [inline src/modules/trading-view/services/trading.view.service.ts:28]: use a type for return

**Review by claudiulataretu — CHANGES_REQUESTED** [review]

File: src/modules/trading-view/dtos/bars.response.ts (hunk 1 above)
Comment (mad2sm0key) [reply src/modules/trading-view/dtos/bars.response.ts:30]: Added in https://github.com/multiversx/mx-exchange-service/pull/1386

File: src/modules/trading-view/services/trading.view.service.ts (hunk 2 above)
Comment (mad2sm0key) [reply src/modules/trading-view/services/trading.view.service.ts:28]: Added in https://github.com/multiversx/mx-exchange-service/pull/1386

**Pull request closed without merging**